src/datasources/asset/components/AssetQueryEditor.tsx

```diff
@@ -0,0 +1,156 @@
+
+type Props = QueryEditorProps<AssetDataSource, AssetQuery>;
+
+export function AssetQueryEditor({ query, onChange, onRunQuery, datasource }: Props) {
+
+  query = datasource.prepareQuery(query);
+  const workspaces = useWorkspaceOptions(datasource);
+  const [errorMsg, setErrorMsg] = useState<string | undefined>('');
+  const handleError = (error: Error) => setErrorMsg(parseErrorMessage(error));
+
+  const minionIds = useAsync(() => {
+    let filterString = '';
+    if (query.workspace) {
+      filterString += `workspace = "${query.workspace}"`;
+    }
+    return datasource.querySystems(filterString).catch(handleError);
+  }, [query.workspace]);
+
+  const handleQueryChange = (value: AssetQuery, runQuery: boolean): void => {
+    onChange(value);
+    if (runQuery) {
+      onRunQuery();
+    }
+  };
+
+  const onWorkspaceChange = (item?: SelectableValue<string>): void => {
+    if (item?.value) {
+      // if workspace changed, reset Systems and Assets fields
+      handleQueryChange(
+        { ...query, workspace: item.value, minionIds: [] },
+        // do not run query if workspace not changed
+        query.workspace !== item.value
+      );
+    } else {
+      handleQueryChange({ ...query, workspace: '' }, true);
+    }
+  };
+  const handleMinionIdChange = (items: Array<SelectableValue<string>>): void => {
+    if (items) {
+      handleQueryChange(
+        { ...query, minionIds: items.map(i => i.value!) },
+        // do not run query if minionIds not changed
+        !_.isEqual(query.minionIds, items)
+      );
+    } else {
+      handleQueryChange({ ...query, minionIds: [] }, true);
+    }
+  };
+
+  const getVariableOptions = (): Array<SelectableValue<string>> => {
+    return datasource.templateSrv
+      .getVariables()
+      .map((v) => toOption('$' + v.name));
+  };
+  const loadMinionIdOptions = (): Array<SelectableValue<string>> => {
+    let options: SelectableValue[] = (minionIds.value ?? []).map((system: SystemMetadata): SelectableValue<string> => ({
+        'label': system.alias ?? system.id,
+        'value': system.id,
+        'description': system.state,
+      })
+    )
+    options.unshift(...getVariableOptions());
+
+    return options;
+  }
+
+  return (
+    <div style={{ position: 'relative' }}>
+      <InlineField label="Workspace"
+                   tooltip={tooltips.workspace[EntityType.Asset]}
+                   labelWidth={22}
+                   data-testid={selectors.components.assetPlugin.workspace}>
+        <Select
+          isClearable
+          isLoading={workspaces.loading}
+          onChange={onWorkspaceChange}
+          options={workspaces.value}
+          placeholder="Any workspace"
+          value={query.workspace}
+        />
+      </InlineField>
+      <InlineField label="Systems"
+                   tooltip={tooltips.system[EntityType.Asset]}
+                   labelWidth={22}
+                   data-testid={selectors.components.assetPlugin.system}>
```

Comment:
Why do we need the testid attribute here? Can we use `getByLabelText` like we do here: https://github.com/ni/systemlink-grafana-plugins/blob/0e230a372d46e42326bde813bccf74e24c4ed0a8/src/datasources/tag/components/TagQueryEditor.test.tsx#L17

Reply:
getByLabelText not work with Select component
![image](https://github.com/ni/systemlink-grafana-plugins/assets/44769443/3f2e8686-6fe7-4a5e-8502-34e4a75005b6)

_TestingLibraryElementError: Found a label with the text of: Workspace, however the element associated with this label (div) is non-labellable [https://html.spec.whatwg.org/multipage/forms.html#category-label]. If you really need to label a div, you can use aria-label or aria-labelledby instead._

getByRole

Reply:
best solution I founded is find all combo boxes with getAllByRole() and take them by index

Reply:
I think that's fine for now